client/src/components/_common/Sidebar/Sidebar.jsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { NavLink as RRNavLink } from 'react-router-dom';
+import { Nav, NavItem, NavLink } from 'reactstrap';
+import Icon from '_common/Icon';
+import styles from './Sidebar.module.css';
+
+function isNotEmptyString(props, propName, componentName) {
+  if (!props[propName] || props[propName].replace(/ /g, '') === '') {
+    return new Error(`No text passed to ${componentName}. Validation failed.`);
+  }
+  return null;
+}
+
+const SidebarItem = ({ to, iconName, children, disabled }) => {
+  return (
+    <NavItem>
+      <NavLink
+        tag={RRNavLink}
+        to={to}
+        disabled={disabled}
+        className={styles['link']}
+        activeClassName={styles['link--active']}
+      >
+        <div className={`${disabled && styles['disabled']}`}>
+          <Icon name={iconName} />
+          <span className={styles['text']}>{children}</span>
+        </div>
+      </NavLink>
+    </NavItem>
+  );
+};
+SidebarItem.propTypes = {
+  to: PropTypes.string.isRequired,
+  iconName: PropTypes.string.isRequired,
+  children: isNotEmptyString,
+  disabled: PropTypes.bool,
+};
+SidebarItem.defaultProps = {
+  disabled: false,
+};
+
+const Sidebar = ({ sidebarItems }) => {
+  return (
+    <Nav className={styles['root']} vertical>
```

Comment:
__Unapplied Styles__

The dynamic class created from `.root`—my instances names it `._root_13fpc_1`—is __empty__ ([screenshot]), _thus_ the "composed" styles (background color, padding, width, et cetera) from main Sidebar are not carried over ([screenshot]), _**maybe** because_ [the CSS module is trying to import a SASS file](https://github.com/TACC/Core-Portal/blob/task/FP-86/client/src/components/_common/Sidebar/Sidebar.module.css#L2).

<details><summary>Ideas</summary>

1. Don't use SASS (e.g. `.scss`).﹡

    <details><summary><code>Sidebar.module.css</code> Using CSS Custom Props</summary>

    ```
    .root {
      --width: 215px;
    
      /* ... */
      min-width: var(--width);
      max-width: var(--width);
      width: var(--width);
      /* ... */
    }
    ```

    </details>

2. Move actual styles to `_common/Sidebar`†. Have `Sidebar` compose styles from `_common/Sidebar`.﹡

3. Move actual styles to `_common/Sidebar`†. Have `Sidebar` be built using `_common/Sidebar`.﹡

<sub>﹡ You can, instead of SASS variables, use [native CSS custom properties](https://developer.mozilla.org/en-US/docs/Web/CSS/--*).</sub>
<sub>† And don't use SASS (e.g. `.scss`).﹡</sub>

</details>

[screenshot]: https://user-images.githubusercontent.com/62723358/158688094-73a7e3d7-a533-401a-be6a-276e00dd0187.png

Reply:
For the last two ideas, would that be best implemented in a future "use _common Sidebar" ticket?

Reply:
Best, I dunno. It can be done here or a new ticket; I don't care which. (FP-86 scheduling is less complex than the button component.) For a new ticket, please note what the expected outcome is.